packages/hardhat/contracts/PredictorGame.sol

```diff
@@ -209,5 +229,32 @@ contract PredictorGame {
 
 		emit PlayShort(msg.sender, epoch, amount);
 	}
-    
+
+	// following cei pattern and claiming rewards
+
+	/**
+	 * @dev follows cei pattern
+	 * @notice incomplete, there is a missing peice that is if the winner has won or not which was not implemented yet so once it is done, this will need be modified
+	 */
+	function claimRewards(uint256 epoch) external {
+		//checking if the user has already claimed
+		if (ledger[epoch][msg.sender].claimed) {
+			revert("You have already claimed rewards for this round");
+		}
+		// total rewards
+		uint256 total = rounds[epoch].totalAmount;
+		// other users? - fees of protocol
+		// uint256 rewardForUser = total/<no. of other users> - fees should be in % for protocol;
+
+		// here we can check if the user shorted or longer and have an if clause that would use uni v4 hook
+		// for simplicity i am rewarding them 10% of what is in the pool of the current epoch
+		uint256 rewardForUser = total / 10; // will there be precision loss?
```

Comment:
Will the rewards gained from test networks significantly contribute to the overall development and stability of the project?

Reply:
no i just put this if you wanted to have this system in place, as i sayed it can be edited and thus i left the comments of things that can be improved on